app/src/main/java/com/humberto/tasky/agenda/data/task/TaskRepositoryImpl.kt

```diff
@@ -47,6 +55,30 @@ class TaskRepositoryImpl @Inject constructor(
                             userId = localUserId!!
                         )
                     )
+                    enqueuePendingWorker<CreateTaskWorker>(taskId = task.id)
+                    return Result.Success(Unit)
+                }
+            }
+            return result
+        } catch (e: SQLiteFullException) {
+            Result.Error(DataError.Local.DISK_FULL)
+        }
+    }
+
+    override suspend fun updateTask(task: AgendaItem.Task): EmptyResult<DataError> {
+        return try {
+            val taskEntity = task.toTaskEntity()
+            taskDao.upsertTask(taskEntity)
+            val result = safeCall {
+                agendaApi.updateTask(task.toTaskRequest())
+            }.onError { error ->
+                if (error.isRetryable()) {
+                    taskDao.insertTaskPendingSync(
+                        taskEntity.toTaskPendingSyncEntity(
+                            userId = localUserId!!
+                        )
+                    )
+                    enqueuePendingWorker<UpdateTaskWorker>(taskId = task.id)
```

Comment:
`applicationScope` 👀 
You want to make sure these calls are executed

Reply:
Yes, you can’t guarantee here that the caller will use the `applicationScope` to call this function. Place the call at the source, and don’t assume the caller will offload the entire function to the appScope. You also don’t want to put all your coroutines in the appScope, only use it with a specific purpose in mind.